src/Netscript/APIWrapper.ts

```diff
@@ -107,10 +125,11 @@ function wrapFunction(
     },
   };
   function wrappedFunction(...args: unknown[]): unknown {
+    checkEnv(ctx);
     helpers.updateDynamicRam(ctx.function, getRamCost(Player, ...tree, ctx.function));
     return func(ctx)(...args);
   }
-  const parent = getNestedProperty(wrappedAPI, ...tree);
+  const parent = getNestedProperty(wrappedAPI, tree);
```

Comment:
this should still have the argument spread i think (...tree)

Reply:
With spread on one end and rest on the other, it's taking an array and turning it into individual entries on one end, then converting multiple entries back into an array on the other. I just changed both ends to just pass the `tree` array directly so the end result is the same, there's just less converting back and forth.

Reply:
i can understand that i suppose